pytorch_lightning/trainer/data_loading.py

```diff
@@ -48,15 +48,30 @@ def init_train_dataloader(self, model):
         if EXIST_ITER_DATASET and isinstance(self.get_train_dataloader().dataset, IterableDataset):
             self.num_training_batches = float('inf')
         else:
+            if not 0. <= self.train_percent_check <= 1.:
+                raise ValueError(f"train_percent_check must lie in the range [0.0, 1.0], but got "
+                                 f"{self.train_percent_check:.3f}.")
+
             self.num_training_batches = len(self.get_train_dataloader())
             self.num_training_batches = int(self.num_training_batches * self.train_percent_check)
+            self.num_training_batches = max(1, self.num_training_batches)
```

Comment:
I would consider adding a check and optionaly raise error if the nb is 0 becase with empty dataloader it may crash later anyway...

Reply:
After the #653 fix there should be no problems with empty dataloaders for now. So I decided to remove `max(1, ...)` for training and test batches.

But I agree that maybe we should better think about the cases when when `num_training_batches=0` or `len(train_dataloader)=0`. It is not very straightforward because the user, for example, can define the model with only `test_step` and `test_end` and `test_dataloader` for testing. In that case an empty `train_dataloader` is acceptable.